src/components/refPropTypes.js

```diff
@@ -0,0 +1,6 @@
+import PropTypes from 'prop-types';
+
+export default PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({current: PropTypes.elementType}),
+]);
```

Comment:
Thoughts on this? If we like this, should we refactor all the other places we're passing ref props to also use this in propTypes?

Reply:
I think we should actually not put much more effort into maintaining `propTypes` if we are going to adopt TypeScript